pkg/roll/run_update.go

```diff
@@ -371,6 +371,38 @@ func (u *update) enable() error {
 		return err
 	}
 
+	newRC, err := u.rcs.Get(u.NewRC)
+	if err != nil {
+		return err
+	}
+
+	if newRC.Disabled {
+		// We must exercise caution before enabling a disabled RC.
+		//
+		// Consider a deploy:
+		// Old RC:                        | New RC:
+		// disabled, 2 desired, 2 labeled | enabled, 0 desired, 0 labeled.
+		// disabled, 1 desired, 2 labeled | enabled, 1 desired, 0 labeled.
+		//
+		// Now let's say we interrupt the deploy here,
+		// and because of Consul latency the new RC doesn't get to act on its desire.
+		// Then, we rollback.
+		//
+		//  enabled, 1 desired, 2 labeled | disabled, 1 desired, 0 labeled.
+		//
+		// At this point, the formerly-old RC momentarily unschedules one node, because it has too many.
```

Comment:
Honestly though, you could argue that we could optimize here by simply setting the RCs to 2/0 at this point, if they haven't had a chance to act on their desire. This might get racy if they were in the process of doing so, though.